### PR TITLE
Add support for IPv6 networked containers

### PR DESCRIPTION
**Pull request by spikecurtis**
High level changes:
 - add commands for IPv6 address pools
 - add bird6 to calico-node with Confd configuration
 - move to latest calico-felix (including zeroMQ API update for interface names) for proxy NDP config
 - program containers with local gateway, rather than default via device directly
 - version checking of IP address to program netns correctly

### CODE REVIEW

**Comment by spikecurtis** [comment]
Don't merge this until we get an new Felix release!  The zeroMQ API is changed, so it will not work with current Felix.

File: calicoctl.py [hunk 1]
```diff
@@ -228,6 +235,10 @@ def node(ip, node_image):
     # Set up etcd
     client = DatastoreClient()
 
+    # Enable IP forwarding.
```
Comment (Symmetric) [inline calicoctl.py]: Might be beneficial to future generations to describe what we're enabling forwarding for.

File: calicoctl.py [hunk 2]
```diff
@@ -237,9 +248,13 @@ def node(ip, node_image):
     if not ipv4_pools:
         print "No IPv4 range defined.  Exiting."
         return
+    ipv6_pools = client.get_ip_pools("v6")
+    if not ipv6_pools:
```
Comment (Symmetric) [inline calicoctl.py]: Is it an error to have no ipv6_pools if we're not configured with an IPv6 address? Looks like we bail in that case right now.

Comment (Symmetric) [inline calicoctl.py]: Just spotted that we default the pools in master() below. Probably worth a comment here to explicitly call out that we're defaulting the pools elsewhere.

File: master/root_overlay/plugin.py [hunk 3]
```diff
@@ -185,6 +185,7 @@ def do_ep_api():
             rsp = {"rc": "SUCCESS",
                    "message": "Hooray",
                    "type": fields['type'],
+                   "interface_prefix": "tap",
```
Comment (Symmetric) [inline master/root_overlay/plugin.py]: Comment this? Unclear why it's being added.

File: master/root_overlay/plugin.py [hunk 4]
```diff
@@ -242,9 +243,11 @@ def send_all_eps(create_sockets, host, resync_id):
     # Send all of the ENDPOINTCREATED messages.
     for ep in eps_by_host.get(host, {}):
         log.info("Sending ENDPOINTCREATED message for endpoint %s", ep)
+        if_name = "tap" + ep[:11]
```
Comment (Symmetric) [inline master/root_overlay/plugin.py]: What's 11?

File: node/root_overlay/adapter/netns.py [hunk 5]
```diff
@@ -60,18 +64,25 @@ def remove_endpoint(ep_id):
...
+
+def set_up_endpoint(ip, cpid, next_hop_ips,
+                    in_container=False,
+                    veth_name=VETH_NAME,
+                    proc_alias=PROC_ALIAS):
     """
     Set up an endpoint (veth) in the network namespace idenfitied by the PID.
 
-    :param ip: The IP address to assign to the endpoint (veth)
+    :param ip: The IP address to assign to the endpoint (veth) as Netaddr IPAddress.
```
Comment (Symmetric) [inline node/root_overlay/adapter/netns.py]: Much though it pains me to make this complaint, these lines are >79 chars long.

**Comment by Symmetric** [comment]
@spikecurtis - some doc/comment/formatting suggestions, but functionally LGTM.

**Pull request merged**